examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -324,6 +331,7 @@ export class TaskList extends DataObject implements ITaskList {
         this._draftData = await draft.get();
 
         this._draftData.on("valueChanged", (changed) => {
+            this._fluidSync = false;
```

Comment:
This might be an awkward place to detect the sync state - you need to distinguish whether the draft data is changing from:
1. A task is being added/deleted locally (using the Fluid app) and so it should result in an "unsaved draft change"
2. A task is being added/deleted from the draft data as a result of re-sync'ing with the external data (because the task was added/deleted in the external UI) and no "unsaved draft change".

You may need Kajari's work to be able to inspect the `Task` and figure out which case it is.